include/gotcha/gotcha.h

```diff
@@ -105,9 +105,28 @@ GOTCHA_EXPORT enum gotcha_error_t gotcha_set_priority(const char* tool_name, int
  */
 GOTCHA_EXPORT enum gotcha_error_t gotcha_get_priority(const char* tool_name, int *priority);
 
+/*!
+ ******************************************************************************
+ *
+ * \fn enum void* gotcha_get_wrappee(gotcha_wrappee_handle_t)
+ *
+ * \brief Given a GOTCHA wrapper's handle, returns the wrapped function for it to call
+ *
+ * \param handle The wrappee handle to return the function pointer for
+ *
+ ******************************************************************************
+ */
+GOTCHA_EXPORT void* gotcha_get_wrappee(gotcha_wrappee_handle_t handle);
+
 #if defined(__cplusplus) 
 }
 #endif
 
+#if defined(__cplusplus)
+template<typename T,typename DataPointerType>
+T* castFunctionPointer(DataPointerType in){
+  return reinterpret_cast<T*>(in);
+}
+#endif
```

Comment:
This C++ code doesn't seem to add much value, beyond renaming reinterpret_cast.  Users would still have to do the difficult work of getting the type of the new function pointer and say what to cast.  Would suggest dropping this if we can't find a more useful way to manage the function pointers.

Reply:
Let me examine this, for whatever reason I wasn't able to get raw reinterpret_cast to work.

Reply:
Oh, I got it, the way I've written it you get to leave off the ampersand. Think I'm game to give that up. Nice, I was leaning towards a separate "C++ utilities" file that I didn't really want to maintain